### PR TITLE
Enable active series metrics in the ingester by default

### PR DESCRIPTION
**Pull request by 56quarters**
Signed-off-by: Nick Pillitteri <nick.pillitteri@grafana.com>

**What this PR does**:

This change calculates and exports the `cortex_ingester_active_series`
metric by default. Up to this point, the metric was disabled by default since
calculating it consumes some amount of memory.

The original PR (#3153) estimated for 1M active series at least 40MB
and up to another 200MB depending on our luck reusing labels from the
ref cache.

We (Grafana) have been running with this setting enabled on our ingesters
for some time and the resource usage doesn't appear to be significant.
This feature appears to add between **1.2 - 1.6%** in memory usage when
enabled: ~140MB out of a total of ~10GB of memory used per ingester.

The ingesters I measured this on

* Have multiple tenants running production workloads
* Have about 1.3M active series each
* Have about a 10GB working set (as measured by `kubectl top` and exported
  k8s metrics)

Based on this and the utility of the metric itself, I'd like to enable it
by default.

Screenshots of the pprof heap output attached.

table
graph

**Checklist**
~- [ ] Tests updated~
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
   - `-alertmanager.receivers-firewall.block.cidr-networks` renamed to `-alertmanager.receivers-firewall-block-cidr-networks`
   - `-alertmanager.receivers-firewall.block.private-addresses` renamed to `-alertmanager.receivers-firewall-block-private-addresses`
 * [CHANGE] Change default value of `-server.grpc.keepalive.min-time-between-pings` to `10s` and `-server.grpc.keepalive.ping-without-stream-allowed` to `true`. #4168
+* [CHANGE] Ingester: Change default value of `-ingester.active-series-metrics-enabled` to `true`. #4257
```
Comment (stevesg) [inline CHANGELOG.md:10]: Maybe include a very brief FYI-style comment here from your analysis? "This incurs a small increase in memory usage...etc ". The heads-up might be valuable to some people.

**Review by stevesg — APPROVED** [review]
Appreciate the detailed analysis. Thanks!

One optional nit.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Review by pstibrany — APPROVED** [review]
LGTM, thank you.

> Up to this point, the metric was disabled by default since calculating it consumes some amount of memory.

That's not the only reason. Main reason is that everybody does a calculation of active series differently.

This feature was introduced because after switch from chunks to blocks engine, "in memory series" started to be computed differently. To make it more configurable and compatible with computation in chunks storage, we have introduced "active series" metrics.

**Pull request merged**